pkg/declarativeconfig/auth_provider.go

```diff
@@ -0,0 +1,67 @@
+package declarativeconfig
+
+// RequiredAttribute is representation of storage.AuthProvider_RequiredAttribute that supports transformation from YAML.
+type RequiredAttribute struct {
+	AttributeKey   string `yaml:"key,omitempty"`
+	AttributeValue string `yaml:"value,omitempty"`
+}
+
+// ClaimMapping represents a single entry in "claim_mappings" field in auth provider proto.
+type ClaimMapping struct {
+	Path string `yaml:"path,omitempty"`
+	Name string `yaml:"name,omitempty"`
+}
+
+// OIDCConfig contains config values for OIDC auth provider.
+type OIDCConfig struct {
+	Issuer                    string `yaml:"issuer,omitempty"`
+	CallbackMode              string `yaml:"mode,omitempty"`
+	ClientID                  string `yaml:"clientID,omitempty"`
+	ClientSecret              string `yaml:"clientSecret,omitempty"`
+	DisableOfflineAccessScope bool   `yaml:"disableOfflineAccessScope,omitempty"`
+	DontUseClientSecretConfig bool   `yaml:"dontUseClientSecretConfig,omitempty"`
+}
+
+// SAMLConfig contains config values for SAML auth provider.
+type SAMLConfig struct {
```

Comment:
Would be also nice to have more documentation about the requirements of the fields, i.e. if `metadataURL` is set, you do not need to provide the `ssoURL, nameIdFormat`.